src/sam/mod.rs

```diff
@@ -0,0 +1,175 @@
+use std::ffi;
+use std::path::Path;
+
+use htslib;
+
+use bam::header;
+use bam::record;
+use bam::Reader;
+use bam::Read;
+
+// new on bam::HeaderView is not public
+pub struct HeaderView {
+    inner: *mut htslib::bam_hdr_t,
```

Comment:
yes, but some methods on it must be made public. this really falls under refactoring out common traits from bam+sam writers. i have made the appropriate methods in bam module public (i suspect that is what is causing the conflicts)